intel/init.py

```diff
@@ -39,14 +39,83 @@ def init(conf_dir, num_exclusive_cores, num_shared_cores,
+            sys.exit(1)
+
+        check_sst_cp_isolated_cores(platform, num_exclusive_cores,
+                                    num_shared_cores, epp_order)
+        logging.info("EPP order: {}".format(", ".join(epp_order)))
+
+        if len(epp_order) == 3:
+            sst_cp_exclusive_cores = platform.get_epp_cores(
+                 epp_order[0], num_exclusive_cores)
+            sst_cp_shared_cores = platform.get_epp_cores(
+                 epp_order[1], num_shared_cores)
+            sst_cp_infra_cores = platform.get_epp_cores_no_limit(
+                 epp_order[2])
+
+            logging.info("Isolated SST-CP exclusive cores (EPP value:"
+                         " {}): {}".format(epp_order[0], ",".join(
+                          [str(c.core_id) for c in sst_cp_exclusive_cores])))
+            logging.info("Isolated SST-CP shared cores (EPP value: {}"
+                         "): {}".format(epp_order[1], ",".join(
+                          [str(c.core_id) for c in sst_cp_shared_cores])))
+            logging.info("SST-CP infra cores (EPP value: {}): {}"
+                         .format(epp_order[2], ",".join([str(c.core_id)
+                                 for c in sst_cp_infra_cores])))
+        elif len(epp_order) == 2:
+            logging.info("Only two EPP values set; exclusive and "
+                         "shared pools will take all of the cores "
+                         "of the highest EPP value")
+            sst_cp_exclusive_cores = platform.get_epp_cores(
+                 epp_order[0], num_exclusive_cores)
+            sst_cp_shared_cores = platform.get_epp_cores(
+                 epp_order[0], num_shared_cores, sst_cp_exclusive_cores)
+            sst_cp_infra_cores = platform.get_epp_cores_no_limit(
+                 epp_order[1])
+
+            logging.info("Isolated SST-CP exclusive cores (EPP value: {}): {}"
+                         .format(epp_order[0], ",".join([str(c.core_id)
+                                 for c in sst_cp_exclusive_cores])))
+            logging.info("Isolated SST-CP shared cores (EPP value: {}): {}"
+                         .format(epp_order[0], ",".join([str(c.core_id)
+                                 for c in sst_cp_shared_cores])))
+            logging.info("SST-CP infra cores (EPP value {}): {}"
+                         .format(epp_order[1], ",".join([str(c.core_id)
+                                 for c in sst_cp_infra_cores])))
+        else:
+            logging.error("There must be either 2 or 3 EPP values set among "
+                          "the cores. EPP values present: {}".format(
+                              ", ".join(epp_order)))
+            sys.exit(1)
+
+        assign(sst_cp_exclusive_cores, "exclusive",
+               count=num_exclusive_cores)
+        assign(sst_cp_shared_cores, "shared",
+               count=num_shared_cores)
+        assign(sst_cp_infra_cores, "infra")
+
+    elif sst_bf and platform.has_isolated_sst_bf_cores():
```

Comment:
Quick question - I'm assuming that are SST-BF and SST-CP mutually exclusive? With this implementation, SST-BF won't be discovered if SST-CP is enabled. If that's how it's supposed to work, then we should limit the configuration options and ensure that sst_bf and sst_cp are not enabled at the same time in the checks here in lines 35 and 42.

Reply:
or at least just throw a warning so the end user doesn't have to scratch their head wondering why SST-BF isn't used

Reply:
Originally it was going to be NFD that wouldn't label a node with SST-CP enabled if SST-BF was enabled, but this has been changed so both features can be labeled as enabled so will add this check and fail.